taichi/program/ndarray.cpp

```diff
@@ -14,6 +14,10 @@ namespace {
 
 size_t flatten_index(const std::vector<int> &shapes,
                      const std::vector<int> &indices) {
+  if (shapes.size() == 0) {
+    TI_ASSERT(indices.size() == 1 && indices[0] == 0);
+    return 0;
+  }
   TI_ASSERT(shapes.size() == indices.size());
   if (indices.size() == 1) {
```

Comment:
Why do we need this if-else? IIUC we can write
```
size_t ind = 0;
for (int i = 0; i < indices.size(); i++) {
  ind = ind * shapes[i] + indices[i];
}
```
and then we don't need to care about the size of `indices`, and it works if the size is 0.

Reply:
lol good point, let me update